### ISSUE TITLE
Use docker image for dependencies in CI jobs

### ISSUE THREAD

**Issue by PhilMiller** [ECP-WarpX/WarpX#3108]
Each CI job on Azure spends 5-10 minutes in the 100%-repetitive "Install dependencies" step, in which packages are installed from apt and libraries are rebuilt from source. If a docker image were built with these installations completed, the CI jobs could refer to it, and save the time spent doing this in every job.

**Comment by PhilMiller** [comment]
Python packages could be pre-installed, too. Even if they're not all the versions that the codebase itself requests, many will suffice and so fewer will be installed later.

**Comment by ax3l** [comment]
Agreed. My main question would be how we build the containers: shall we spin up a separate repo in `ECP-WarpX/`, place the `Dockerfiles` there and use the GitHub container registry (maybe traffic limited) or public docker hub?
